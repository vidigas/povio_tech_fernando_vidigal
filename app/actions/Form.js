	export const onUserChange = (payload) => {
		return {
			type: 'CHANGE_USERNAME',
			payload: payload.target.value
		};
	};


	export const onPasswordChange = (payload) => {
		return {
			type: 'CHANGE_PASSWORD',
			payload: payload.target.value
		};
	};
