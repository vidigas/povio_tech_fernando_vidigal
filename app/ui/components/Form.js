import React from 'react';
import injectSheet from 'react-jss';
import propTypes from 'prop-types';


import TextInput from '../atoms/TextInput';

const Form = ({ classes, fields, handleSubmit}) => {
	var arr = fields.map( field => {
		
	})

	return (
		)
}