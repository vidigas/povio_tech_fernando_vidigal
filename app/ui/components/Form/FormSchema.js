const login_form_schema = {
	title: 'LOGIN',
	fields : {
		username : {
			placeholder: 'username',
			type: 'text'
		},
		password: {
			placeholder: 'password',
			type: 'password'
		}
	},
	buttons: {
		button_send: {
			title: 'login'
		},
		button_link: {
			title: 'notRegistered'
		}
	}
}

const signup_form_schema = {
	title: 'SIGNUP',
	fields : {
		username : {
			placeholder: 'username',
			type: 'text'
		},
		password: {
			placeholder: 'password',
			type: 'password'
		}
	},
	buttons: {
		button_send: {
			title: 'register'
		},
		button_link: {
			title: 'login'
		}
	}
}

const update_form_schema = {
	title: 'UPDATE PASSWORD',
	fields : {
		username : {
			placeholder: 'username',
			type: 'text'
		},
		oldPassword: {
			placeholder: 'old password',
			type: 'password'
		},
		newPassword: {
			placeholder: 'new password',
			type: 'password'
		}
	},
	buttons: {
		button_send: {
			title: 'update'
		}
	}
}

const FormSchema = (form_name) => {
	switch(form_name){
		case 'login': return login_form_schema;
		case 'signup': return signup_form_schema;
		case 'update': return update_form_schema;
		default: return;
	}
}

export default FormSchema;
