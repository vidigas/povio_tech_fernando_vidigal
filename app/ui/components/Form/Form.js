import React from 'react';
import injectSheet from 'react-jss';
import propTypes from 'prop-types';

import Button from '../../atoms/Button';
import TextInput from '../../atoms/TextInput';
import FormSchema from './FormSchema';

const renderFields = (fieldsObj, onPasswordChange, onUserChange) => {
	return Object.keys(fieldsObj).map( fieldName => {
		const onChange = fieldName === 'username' ? onUserChange : onPasswordChange; 
		return <TextInput type={fieldName === 'password' ? 'password' : 'text'} label={fieldName} onChange={onChange} key={fieldName}/>;
	});
}

const selectHandler = (handlers, title) => {
	switch(title){
		case 'login': {
			return 
		}
		default: return
	}
}

const renderButtons = (buttonsObj, onPrimaryButtonClick, onSecundaryButtonClick) => {
	return Object.keys(buttonsObj).map( buttonName => {

		const onClick = buttonName === 'button_send' ? onPrimaryButtonClick : onSecundaryButtonClick;
		return <Button label={buttonsObj[buttonName].title} onClick={onClick} key={buttonsObj[buttonName].title} />;
	});

}


const FormBuilder = ({ classes, context, onUserChange, onPasswordChange, onPrimaryButtonClick, onSecundaryButtonClick }) => {
	
	var form_schema = FormSchema(context)
	let fieldsObj = form_schema.fields;
	let buttonsObj = form_schema.buttons;
	return (
		[ <h1 key={'title'}> { form_schema.title } </h1>,
		renderFields(fieldsObj, onPasswordChange, onUserChange),
		renderButtons(buttonsObj, onPrimaryButtonClick, onSecundaryButtonClick)
		])
}


const style = {
	container: {
		backgroundColor: 'white',
		padding: '20px',
		marginTop: '20px'
	}
}

export default injectSheet(style)(FormBuilder);