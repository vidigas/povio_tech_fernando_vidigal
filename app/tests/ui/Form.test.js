import React from 'react';
import renderer from 'react-test-renderer';
import Form from '../../ui/components/Form';

const func = () => {};

it('renders correctly', () => {
  const tree = renderer
    .create(<Form 
    					context={'login'}
    					onUserChange={()=> console.log('change')}
    					onPasswordChange={() => console.log('change')}
    					onPrimaryButtonClick={() => console.log('change')}
    					onSecundaryButtonClick={() => console.log('change')} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

