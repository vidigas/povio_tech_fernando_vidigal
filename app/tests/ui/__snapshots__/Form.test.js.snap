// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h1>
     
    LOGIN
     
  </h1>,
  <div
    className="input-field"
  >
    <label>
      username
    </label>
    <input
      className="TextInput-input-0-1-2"
      onChange={[Function]}
      type="text"
    />
  </div>,
  <div
    className="input-field"
  >
    <label>
      password
    </label>
    <input
      className="TextInput-input-0-1-2"
      onChange={[Function]}
      type="password"
    />
  </div>,
  <button
    className="Button-button-0-1-3 Button-button-0-1-5 "
    onClick={[Function]}
    type="button"
  >
    login
  </button>,
  <button
    className="Button-button-0-1-3 Button-button-0-1-6 "
    onClick={[Function]}
    type="button"
  >
    go to Signup
  </button>,
]
`;
